### ISSUE TITLE
Tab navigation is broken?

### ISSUE THREAD

**Issue by smmalis37** [linebender/druid#1177]
Copying a conversation from zulip:

> Steven: I'm trying to add tab navigation to my program, so I added code to call register_for_focus on the WidgetAdded event as the docs recommended. However after doing so pressing tab still does nothing. Is there more I need to do? I added a print to verify that the call is definitely happening. commit making the change: https://github.com/smmalis37/sudoku_rust/commit/468ec62b5fb215822abc592a1d88b52bca228e05

> Raph Levien: From what you said, it sounds like you're doing the right thing. I think it just needs some digging into what's actually happening.

> Raph Levien: I think tab navigation might be broken, I'm seeing it not work in the examples :/

> Raph Levien: For example, in styled_text, it unfocuses the TextBox it's in, but doesn't focus the next one.

I couldn't find an issue tracking fixing tab focusing, so I made one.

**Comment by cmyr** [comment]
yes this broke sometime in the past month or so, I'm not sure exactly what's going on.

**Comment by smmalis37** [comment]
It seems like calling focus_prev and focus_next myself is working (responding to KeyDown left and right arrows), but automatic tab handling is not. I could just add my own handling for Tab and Shift+Tab I suppose.

**Comment by smmalis37** [comment]
Some notes from trying to dig into this:

Textbox is handling tab navigation manually at https://github.com/linebender/druid/blob/99b9e7bd97a01b83bcbbc7950dd22308e540e121/druid/src/widget/textbox.rs#L331
Since it also unconditionally calls register_for_focus at https://github.com/linebender/druid/blob/99b9e7bd97a01b83bcbbc7950dd22308e540e121/druid/src/widget/textbox.rs#L372 it probably shouldn't manually handle tabs too? No other widget has this manual behavior, however no other widget registers itself for focus either.

Doing a global search for "tab" finds the aforementioned textbox handling, and plenty of keyboard translation code, but nobody else listening for it. Did that code just accidentally disappear or am I missing something?

A search for focus_chain also turns up nothing looking for tabs. There's this bit handling calls to prev/next https://github.com/linebender/druid/blob/1e28dcf55bca729232962c3b4a2b921c4adbc959/druid/src/window.rs#L240
but nothing else I can find.

**Comment by cmyr** [comment]
Focus is also being reworked in #1089, which will interfere with this.

**Comment by sysint64** [comment]
I have a similar issue. In my case, `Tab` is working on Linux, but `Shift+Tab` not. On Mac OS, navigation works fine.
